src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp

```diff
@@ -475,7 +475,19 @@ class npc_stillpine_capitive : public CreatureScript
                 return;
 
             if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
-                player->KilledMonsterCredit(me->GetEntry(), me->GetGUID());
+            {
+                if (Group* group = player->GetGroup())
```

Comment:
this part is quite similar to `RewardPlayerAndGroupAtEvent` could this not be used?
https://github.com/azerothcore/azerothcore-wotlk/blob/5010236ad501003fc8284a9d2937715016c70380/src/server/game/Entities/Player/Player.cpp#L12734